codespell_lib/tests/test_dictionary.py

```diff
@@ -1,61 +1,152 @@
+    if os.getenv('ALLOW_NO_ASPELL', 'false').lower() != 'true':
+        raise RuntimeError(
+            'Cannot run complete tests without aspell, consider installing '
+            'it, or set ALLOW_NO_ASPELL=true. Got error during import:\n%s'
+            % (exp,))
+
+ws = re.compile(r'.*\s.*')  # whitespace
+comma = re.compile(r'.*,.*')  # comma
+
+
+# Filename, should be seen as errors in aspell or not
+_data_dir = op.join(op.dirname(__file__), '..', 'data')
+_fnames_in_aspell = [
+    (op.join(_data_dir, 'dictionary%s.txt' % d[2]), d[3:5])
+    for d in _builtin_dictionaries]
+fname_params = pytest.mark.parametrize('fname, in_aspell', _fnames_in_aspell)
+
+
+def test_dictionaries_exist():
+    """Test consistency of dictionaries."""
+    doc_fnames = set(op.basename(f[0]) for f in _fnames_in_aspell)
+    got_fnames = set(op.basename(f)
+                     for f in glob.glob(op.join(_data_dir, '*.txt')))
+    assert doc_fnames == got_fnames
+
+
+@fname_params
+def test_dictionary_formatting(fname, in_aspell):
     """Test that all dictionary entries are valid."""
-    err_dict = dict()
-    ws = re.compile(r'.*\s.*')  # whitespace
-    comma = re.compile(r'.*,.*')  # comma
-    with open(op.join(op.dirname(__file__), '..', 'data',
-                      'dictionary.txt'), 'rb') as fid:
+    errors = list()
+    with open(fname, 'rb') as fid:
         for line in fid:
             err, rep = line.decode('utf-8').split('->')
             err = err.lower()
             rep = rep.rstrip('\n')
-            assert err != rep.lower(), 'error %r corrects to itself' % err
+            try:
+                _check_err_rep(err, rep, in_aspell, fname)
+            except AssertionError as exp:
+                errors.append(str(exp).split('\n')[0])
+    if len(errors):
+        raise AssertionError('\n' + '\n'.join(errors))
+
+
+def _check_aspell(word, msg, in_aspell, fname):
+    if speller is None:
+        return  # cannot check
+    if in_aspell is None:
+        return  # don't check
+    if ' ' in word:
+        return  # can't check (easily)
+    this_in_aspell = speller.check(
+        word.encode(speller.ConfigKeys()['encoding'][1]))
+    end = 'be in aspell for dictionary %s' % (fname,)
+    if in_aspell:  # should be an error in aspell
+        assert this_in_aspell, '%s should %s' % (msg, end)
+    else:  # shouldn't be
+        assert this_in_aspell, '%s should not %s' % (msg, end)
+
+
+def _check_err_rep(err, rep, in_aspell, fname):
+    assert ws.match(err) is None, 'error %r has whitespace' % err
+    assert comma.match(err) is None, 'error %r has a comma' % err
+    assert len(rep) > 0, ('error %s: correction %r must be non-empty'
+                          % (err, rep))
+    assert not re.match(r'^\s.*', rep), ('error %s: correction %r '
+                                         'cannot start with whitespace'
+                                         % (err, rep))
+    _check_aspell(err, 'error %r' % (err,), in_aspell[0], fname)
```

Comment:
It can even just be written as `err` in this case, but after many uses of `%` I've come to prefer always embedding the options in a tuple -- it makes it easier to add things later, and avoids problems where the object can end up being a tuple at some point (and hence breaking the `%` operation)

Reply:
Will the brackets alone not be sufficient for the tuple, or do you need the extra trailing comma so it doesn't get interpreted as something else?

Reply:
Without the comma the parentheses don't turn it into a tuple. Creating a single element tuple requires using a trailing comma (to differentiate it from just being a parenthetical expression)

Reply:
Okay I'll fix the one I "tidied".

Reply:
99783123b50c9f9a812f26e60ed6fea9e3cb96f4